src/annotator/highlighter.js

```diff
@@ -161,17 +161,37 @@ function wholeTextNodesInRange(range) {
     return [];
   }
 
-  const textNodes = [];
+  /** @type {Node|null} */
+  let root = range.commonAncestorContainer;
+  if (root.nodeType !== Node.ELEMENT_NODE) {
```

Comment:
Won't this throw (`TypeError`) if `root` is `null`? It's curious that TS wouldn't complain about that so I feel like I'm missing something.

Edited: Apparently, the value won't be `null` after that first assignment, but could be `null` after the `root.parentElement` assignment. Would be helpful to have a comment, perhaps?